contrib/twitchtv/twirp/twirp.go

```diff
@@ -0,0 +1,208 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+// Package twirp provides tracing functions for tracing clients and servers generated
+// by the twirp framework (https://github.com/twitchtv/twirp).
+package twirp // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/twitchtv/twirp"
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"net/http"
+	"os"
+	"strconv"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/twitchtv/twirp"
+)
+
+type contextKey int
+
+const (
+	twirpErrorKey contextKey = iota
+)
+
+// HTTPClient is duplicated from twirp's generated service code.
```

Comment:
Maybe I'm lacking context here, but it seems a bit counterproductive to include this interface only in generated code. Should we open an issue over on their end? Seems like bad architecture. The only situation where I can see this make sense is if the interface is not constant, but it is: 

https://github.com/twitchtv/twirp/blob/677ba1a47ae278934e5911498e65f917c845dc69/protoc-gen-twirp/generator.go#L344-L346

No change needed on your behalf, just wondering...

Reply:
The `HTTPClient` and [`TwirpServer`](https://github.com/twitchtv/twirp/blob/677ba1a47ae278934e5911498e65f917c845dc69/protoc-gen-twirp/generator.go#L348-L371) interfaces would be useful to have in the `twirp` package instead of only in the generated twirp service code.

I suspect it's intentional, so that new methods can be added without breaking existing code.

Reply:
Yeah, it would be idea if we could import it. But I guess this is fine for now. If it starts getting messy, we can reassess.